src/mlpack/methods/reinforcement_learning/q_networks/categorical_dqn.hpp

```diff
@@ -0,0 +1,246 @@
+/**
+ * @file methods/reinforcement_learning/q_networks/categorical_dqn.hpp
+ * @author Nishant Kumar
+ *
+ * This file contains the implementation of the categorical deep q network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_CATEGORICAL_DQN_HPP
+#define MLPACK_METHODS_RL_CATEGORICAL_DQN_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/loss_functions/mean_squared_error.hpp>
+#include <mlpack/methods/ann/loss_functions/empty_loss.hpp>
+#include "../training_config.hpp"
+
+namespace mlpack {
+namespace rl {
+
+using namespace mlpack::ann;
+
+/**
+ * Implementation of the Distributional Deep Q-Learning network.
```

Comment:
```suggestion
 * Implementation of the Categorical Deep Q-Learning network.
```

Reply:
Both the words are synonyms in this context. In fact, the paper is named distributional. But I'm ok to change it if u say?

Reply:
I think it makes sense to be consistent throughout the documentation.